lxd/profiles.go

```diff
@@ -40,39 +37,49 @@ var profileCmd = Command{
 
 /* This is used for both profiles post and profile put */
 func profilesGet(d *Daemon, r *http.Request) Response {
-	results, err := d.cluster.Profiles()
-	if err != nil {
-		return SmartError(err)
-	}
+	project := projectParam(r)
 
 	recursion := util.IsRecursionRequest(r)
 
-	resultString := make([]string, len(results))
-	resultMap := make([]*api.Profile, len(results))
-	i := 0
-	for _, name := range results {
-		if !recursion {
-			url := fmt.Sprintf("/%s/profiles/%s", version.APIVersion, name)
-			resultString[i] = url
-		} else {
-			profile, err := doProfileGet(d.State(), name)
+	var result interface{}
+	err := d.cluster.Transaction(func(tx *db.ClusterTx) error {
+		hasProfiles, err := tx.ProjectHasProfiles(project)
+		if err != nil {
+			return errors.Wrap(err, "Check project features")
+		}
+
+		if !hasProfiles {
+			project = "default"
+		}
```

Comment:
So if the project doesn't have profiles it is the default project?

Reply:
I'm assuming ProjectHasProfiles hear check whether the project has the features.profiles feature enabled.
If it doesn't, then the project gets its profiles from the default project, so the logic looks correct to me.

Reply:
Correct, ```hasProfiles``` checks features.profiles.